### PR TITLE
Syphilis test is missing in Tests tab even though it was skipped in Tasks

### PR DESCRIPTION
**Pull request by dubdabasoduba**

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 50
- Added 1
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/adapter/LibraryContentAdapter.java  4
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/LibraryContentActivity.java  4
- opensrp-anc/src/test/java/org/smartregister/anc/library/repository/ContactTasksRepositoryTest.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PreviousContactRepository.java  9
- opensrp-anc/src/main/java/org/smartregister/anc/library/model/LibraryContent.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/viewholder/LibraryContentViewHolder.java  1
- opensrp-anc/src/test/java/org/smartregister/anc/library/sync/BaseAncClientProcessorForJavaTest.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/ProfileTasksFragment.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/presenter/ContactPresenter.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/listener/LibraryContentClickListener.java  3
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/BaseContactActivity.java  8
- opensrp-anc/src/test/java/org/smartregister/anc/library/repository/PatientRepositoryTest.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PatientRepository.java  5
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PartialContactRepository.java  7
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/ContactTasksRepository.java  7
- opensrp-anc/src/main/java/org/smartregister/anc/library/model/ContactVisit.java  6
- opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/LibraryFragment.java  2
- opensrp-anc/src/test/java/org/smartregister/anc/library/activity/MainContactActivityTest.java  3
         

Complexity decreasing per file
==============================
+ opensrp-anc/src/main/java/org/smartregister/anc/library/util/JsonFormUtils.java  -1
         

Clones added
============
- opensrp-anc/src/test/java/org/smartregister/anc/library/repository/ContactTasksRepositoryTest.java  22
- opensrp-anc/src/test/java/org/smartregister/anc/library/fragment/ProfileOverviewFragmentTest.java  2
- opensrp-anc/src/main/java/org/smartregister/anc/library/repository/PatientRepository.java  2
- opensrp-anc/src/test/java/org/smartregister/anc/library/util/JsonFormUtilsTest.java  13
- opensrp-anc/src/test/java/org/smartregister/anc/library/fragment/ProfileTasksFragmentTest.java  12
- opensrp-anc/src/test/java/org/smartregister/anc/library/activity/MainContactActivityTest.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

File: opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java [hunk 1]
```diff
@@ -1,17 +1,27 @@
 package org.smartregister.anc.library.interactor;
 
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+
+import org.apache.commons.lang3.StringUtils;
```
Comment (keymane) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java]: Codacy Issue found: [Avoid unused imports such as 'org.apache.commons.lang3.StringUtils'](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

File: opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java (hunk 1 above)
Comment (keymane) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java]: Codacy Issue found: [Avoid unused imports such as 'org.apache.commons.lang3.StringUtils'](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

File: opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/AdvancedSearchFragment.java [hunk 2]
```diff
@@ -250,6 +250,38 @@ public void recalculatePagination(AdvancedMatrixCursor matrixCursor) {
...
+            cursor = commonRepository().rawCustomQueryForAdapter(query);
+            cursor.moveToFirst();
+            clientAdapter.setTotalcount(cursor.getInt(0));
+            Log.v("total count here", "" + clientAdapter.getTotalcount());
+
+            clientAdapter.setCurrentlimit(20);
+            clientAdapter.setCurrentoffset(0);
+
+        } catch (Exception e) {
+            Log.e(getClass().getName(), e.toString(), e);
```
Comment (bennsimon) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/AdvancedSearchFragment.java]: use timber

File: opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/AdvancedSearchFragment.java [hunk 3]
```diff
@@ -250,6 +250,38 @@ public void recalculatePagination(AdvancedMatrixCursor matrixCursor) {
...
+
+            sqb.addCondition(filters);
+            query = sqb.orderbyCondition(Sortqueries);
+            query = sqb.Endquery(query);
+
+            Log.i(getClass().getName(), query);
+            cursor = commonRepository().rawCustomQueryForAdapter(query);
+            cursor.moveToFirst();
+            clientAdapter.setTotalcount(cursor.getInt(0));
+            Log.v("total count here", "" + clientAdapter.getTotalcount());
```
Comment (bennsimon) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/AdvancedSearchFragment.java]: use timber

File: opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java [hunk 4]
```diff
@@ -173,11 +165,36 @@ private void initializeMainContactContainers() {
...
+                process(contactForms);
+
+                setRequiredFields(contact);
+                view.setTag(contact);
+                super.startForms(view);
+            }
+        } catch (IOException e) {
+            Timber.e(e, " --> startForms");
+        } catch (Exception e) {
+            Timber.e(e, " --> startForms");
```
Comment (bennsimon) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java]: You are catching both `Exception` and `IOException` one is the child of another, kindly fix it

File: opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/AdvancedSearchFragment.java [hunk 5]
```diff
@@ -250,6 +250,38 @@ public void recalculatePagination(AdvancedMatrixCursor matrixCursor) {
...
+
+        try {
+            SmartRegisterQueryBuilder sqb = new SmartRegisterQueryBuilder(countSelect);
+            String query = "";
+
+            sqb.addCondition(filters);
+            query = sqb.orderbyCondition(Sortqueries);
+            query = sqb.Endquery(query);
+
+            Log.i(getClass().getName(), query);
```
Comment (bennsimon) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/AdvancedSearchFragment.java]: use timber

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)

File: opensrp-anc/src/main/java/org/smartregister/anc/library/model/ContactVisit.java [hunk 6]
```diff
@@ -333,6 +244,47 @@ private void processFormFieldKeyValues(String baseEntityId, JSONObject object, S
 
     }
 
+    private void processTasks(JSONObject formObject) {
```
Comment (ellykits) [inline opensrp-anc/src/main/java/org/smartregister/anc/library/model/ContactVisit.java]: Could you consider splitting this method further. Its hard to follow allow with the so many nested if statements. I guess they are necessary but refactoring some to other separate methods will make it more readable and reduce the complexity.

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java  1
- opensrp-anc/src/main/java/org/smartregister/anc/library/interactor/ProfileFragmentInteractor.java  7
         

Complexity decreasing per file
==============================
+ opensrp-anc/src/main/java/org/smartregister/anc/library/model/ContactVisit.java  -9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-anc/pullRequest?prid=5090621&bid=16746088)
